src/transformers/models/llama/modeling_llama.py

```diff
@@ -103,7 +103,7 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None):
 
     def forward(self, x, position_ids, seq_len=None):
         # x: [bs, num_attention_heads, seq_len, head_size]
-        freqs = (self.inv_freq[:, None].float().expand(-1, position_ids.shape[0]) @ (position_ids.float())).t()
+        freqs = torch.einsum("i,bl->bli", self.inv_freq.float(), position_ids.float())
```

Comment:
No einsum should not be used, for 1 we like to have explicit computation, and secondly, this will fail with different dtypes. Bfloat float etc.